DropDownBox: Add Value Type Notes for Multiple Selection (#8589)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxDropDownBox/1 Configuration/value.md

---
id: dxDropDownBox.Options.value
type: any
default: null
---
---
##### shortDescription

> Specifies the currently selected value. May be an object if [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, the store [key](/api-reference/30%20Data%20Layer/Store/1%20Configuration/key.md '/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key') is specified, and [valueExpr]({basewidgetpath}/Configuration/#valueExpr) is not set.

Reviewer: ```suggestion
Specifies the selected value. This can be an object if [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, the store [key](/api-reference/30%20Data%20Layer/Store/1%20Configuration/key.md '/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key') is specified, and [valueExpr]({basewidgetpath}/Configuration/#valueExpr) is not set.
```

---

> When [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, you should define [valueExpr]({basewidgetpath}/Configuration/#valueExpr) to correctly identify data items. If **valueExpr** is not specified, the component compares object references to display an item, not object values. For instance, if you define the **value** property as an object containing identical values to **dataSource** to select all items, the component displays nothing.

Reviewer: ```suggestion
When [dataSource]({basewidgetpath}/Configuration/#dataSource) contains objects, define [valueExpr]({basewidgetpath}/Configuration/#valueExpr) to identify data items correctly. If **valueExpr** is not specified, the component compares object references instead of object values to determine which item to display. For example, if you set the **value** property to an object whose values match those in **dataSource** to select all items, the component does not display anything.
```

> [note] If you synchronize **value** with the embedded component's **selectedRowKeys**/**selectedItemKeys** property (for instance, [DataGrid.selectedRowKeys](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectedRowKeys)), ensure the type of **value** matches the **selectedRowKeys**/**selectedItemKeys** type (such as [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)).

Reviewer: ```suggestion
[note] If you synchronize **value** with the embedded component's **selectedRowKeys**/**selectedItemKeys** property (for instance, [DataGrid.selectedRowKeys](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectedRowKeys)), ensure the **value** type matches the **selectedRowKeys**/**selectedItemKeys** type (such as [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)).
```
